velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -110,13 +110,25 @@ Expected<int32_t> SparkCastHooks::castStringToDate(
       removeWhiteSpaces(dateString), util::ParseMode::kSparkCast);
 }
 
+StringView trimNumericSuffix(const StringView& v) {
+  if (!v.empty()) {
+    char last = *(v.end() - 1);
+    if (last == 'f' || last == 'F' || last == 'd' || last == 'D') {
+      return StringView(v.data(), v.size() - 1);
+    }
+  }
+  return v;
+}
+
 Expected<float> SparkCastHooks::castStringToReal(const StringView& data) const {
-  return util::Converter<TypeKind::REAL>::tryCast(data);
+  auto trimmed = trimNumericSuffix(data);
+  return util::Converter<TypeKind::REAL>::tryCast(trimmed);
 }
 
 Expected<double> SparkCastHooks::castStringToDouble(
     const StringView& data) const {
-  return util::Converter<TypeKind::DOUBLE>::tryCast(data);
+  auto trimmed = trimNumericSuffix(data);
+  return util::Converter<TypeKind::DOUBLE>::tryCast(trimmed);
```

Comment:
Please update this change to common behavior, presto also aligns with this behavior
```
presto:default> select cast('123.0d' as double);
 _col0 
-------
 123.0 
(1 row)

Query 20251118_103153_00007_2tw34, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
[Latency: client-side: 67ms, server-side: 51ms] [0 rows, 0B] [0 rows/s, 0B/s]
```